# Analyse geographique: dataset geolocation

---

In [24]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

In [25]:
geolocation = pd.read_csv("csv/olist_geolocation_dataset.csv", dtype={'geolocation_zip_code_prefix' : str, 'geolocation_lat' : float, 'geolocation_lng' : float})

In [26]:
geolocation.head(1)

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                       01037       -23.545621       -46.639292   

  geolocation_city geolocation_state  
0        sao paulo                SP

In [27]:
geolocation['geolocation_zip_code_prefix'].value_counts().describe()

count    19015.000000
mean        52.598633
std         72.057907
min          1.000000
25%         10.000000
50%         29.000000
75%         66.500000
max       1146.000000
Name: geolocation_zip_code_prefix, dtype: float64

Il y a 19015 differents zipcodes, un zipcode possede 1146 occurences, avec une moyenne d'occurence d'environ 53 valeurs par zipcode. 

In [28]:
geolocation['geolocation_lat'].describe()

count    1.000163e+06
mean    -2.117615e+01
std      5.715866e+00
min     -3.660537e+01
25%     -2.360355e+01
50%     -2.291938e+01
75%     -1.997962e+01
max      4.506593e+01
Name: geolocation_lat, dtype: float64

Le territoire territoire du bresil s'étend sur 4 395 km du Nord au Sud (5°16'20" Nord à 33°44'32" Sud de latitude) (5.27 | -33.74) et 4 319 km d'Est en Ouest (34°47'30" Est à 73°59'32" Ouest de longitude) (-33.79 | -73.99) et couvre trois fuseaux horaires. Je fais donc un traitement sur les donnees de la longitude et de la latitude afin de supprimer les valeurs incoherentes.

In [29]:
geolocation['geolocation_lat'].dtype

dtype('float64')

In [30]:
geolocation = geolocation[geolocation['geolocation_lat'] <= 5.27]
geolocation = geolocation[geolocation['geolocation_lat'] >= -33.74]
geolocation = geolocation[geolocation['geolocation_lng'] <= -33.79]
geolocation = geolocation[geolocation['geolocation_lng'] >= -73.99]

In [31]:
geolocation[geolocation['geolocation_zip_code_prefix'] == '14409']

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
334422                       14409       -20.509897       -47.397866   
334449                       14409       -20.497396       -47.399241   
334459                       14409       -20.510459       -47.399553   
334462                       14409       -20.480940       -47.394161   
334463                       14409       -20.515413       -47.398194   
...                            ...              ...              ...   
336625                       14409       -20.483960       -47.403324   
336650                       14409       -20.483960       -47.403324   
336653                       14409       -20.487255       -47.405925   
336657                       14409       -20.510459       -47.399553   
336670                       14409       -20.500253       -47.400367   

       geolocation_city geolocation_state  
334422           franca                SP  
334449           franca                SP  
334459           franca                SP  
334462           franca                SP  
334463           franca                SP  
...                 ...               ...  
336625           franca                SP  
336650           franca                SP  
336653           franca                SP  
336657           franca                SP  
336670           franca                SP  

[147 rows x 5 columns]

Il y a beaucoup de latitude et de longitude pour le meme zip code donc je vais chercher a faire la moyenne de ces latitudes et longitudes par zipcode.

In [32]:
geolocation['geolocation_zip_code_prefix'].unique()

array(['01037', '01046', '01041', ..., '99910', '99920', '99952'],
      dtype=object)

In [34]:

geo


/tmp/ipykernel_3409/3579805321.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  geo = geolocation.groupby('geolocation_zip_code_prefix')['geolocation_lat', 'geolocation_lng'].median().reset_index()


geolocation_zip_code_prefix  geolocation_lat  geolocation_lng
0                           01001       -23.550381       -46.634027
1                           01002       -23.548551       -46.635072
2                           01003       -23.548977       -46.635313
3                           01004       -23.549535       -46.634771
4                           01005       -23.549612       -46.636532
...                           ...              ...              ...
19005                       99960       -27.953797       -52.029641
19006                       99965       -28.179542       -52.035551
19007                       99970       -28.343273       -51.873734
19008                       99980       -28.388092       -51.846880
19009                       99990       -28.329595       -51.769362

[19010 rows x 3 columns]

On voit qu'il y a 19009 zipcodes differents. 